### PR TITLE
Emails troubleshooting

### PR DESCRIPTION
**Pull request by MarcelGeo**
Added:
- emails are not send section to server troubleshooting
- updates in env variables

### CODE REVIEW

File: src/server/troubleshoot/index.md [hunk 1]
```diff
@@ -17,3 +17,25 @@ Did you get an error that the server is not properly configured?
...
 
+## Emails are not sent
+
+Mergin Maps server is sending emails to users in following cases:
+
+* Email verification
+* Password reset
+* Incoming access request
+* Invitation to workspace (only for <EnterprisePlatformName />)
+
```
Comment (tomasMizera) [inline src/server/troubleshoot/index.md:28]: I'd prefer not mentioning the exact situations when we send emails. These can change and even the following list is not complete (it is missing the "account created" email).

**Review by tomasMizera — COMMENTED** [review]
Hi @alex-cit, can I ask you to fix/rephrase grammar here? Thanks!

File: src/server/administer/environment.md [hunk 2]
```diff
@@ -1,15 +1,15 @@
 # Configure environment
 
-There are several application settings which can be changed via <GitHubRepo desc="config variables" id="MerginMaps/server/blob/master/.prod.env" />. Some of them have defaults and some of them need to be modified, particularly those with `fixme` placeholders (marked with asterisks below).
+There are several application settings which can be changed via <GitHubRepo desc="config variables" id="MerginMaps/server/blob/master/.prod.env" />. Some of them have defaults and some of them need to be modified, particularly those with required placeholders (marked with asterisks below).
```
Comment (alex-cit) [inline src/server/administer/environment.md:3]:
```suggestion
There are several application settings which can be changed via <GitHubRepo desc="config variables" id="MerginMaps/server/blob/master/.prod.env" />. Some of them have defaults and some of them need to be modified, particularly those with required placeholders (marked with ⭐️ below).
```

File: src/server/troubleshoot/index.md [hunk 3]
```diff
@@ -17,3 +17,18 @@ Did you get an error that the server is not properly configured?
    
 2. Restart the container with the `MERGIN_BASE_URL` variable
 
+## Emails are not sent
+
+If you are not receiving emails, check that the following environment variables are set correctly:
+
+* `MAIL_DEFAULT_SENDER` is mandatory and should be a valid email address
+* `MAIL_SERVER` should be a valid url to SMTP server
```
Comment (alex-cit) [inline src/server/troubleshoot/index.md:25]:
```suggestion
* `MAIL_SERVER` should be a valid URL to SMTP server
```

File: src/server/troubleshoot/index.md [hunk 4]
```diff
@@ -17,3 +17,18 @@ Did you get an error that the server is not properly configured?
    
 2. Restart the container with the `MERGIN_BASE_URL` variable
 
+## Emails are not sent
+
+If you are not receiving emails, check that the following environment variables are set correctly:
+
+* `MAIL_DEFAULT_SENDER` is mandatory and should be a valid email address
```
Comment (alex-cit) [inline src/server/troubleshoot/index.md:24]:
```suggestion
* `MAIL_DEFAULT_SENDER` needs to be a valid email address
```

File: src/server/troubleshoot/index.md [hunk 5]
```diff
@@ -17,3 +17,18 @@ Did you get an error that the server is not properly configured?
...
+* `MAIL_SERVER` should be a valid url to SMTP server
+* `MAIL_PORT` if you have different port than 587, configure it with this variable
```
Comment (alex-cit) [inline src/server/troubleshoot/index.md:26]:
```suggestion
* `MAIL_PORT` SMTP mail server port is set to `587` by default. Change it to the correct value, if you use a different port.
```

File: src/server/troubleshoot/index.md [hunk 6]
```diff
@@ -17,3 +17,18 @@ Did you get an error that the server is not properly configured?
...
+* `MAIL_SUPPRESS_SEND` check if you set it to `False`
```
Comment (alex-cit) [inline src/server/troubleshoot/index.md:27]:
```suggestion
* `MAIL_SUPPRESS_SEND` should be set to `false`
```

File: src/server/troubleshoot/index.md [hunk 7]
```diff
@@ -17,3 +17,18 @@ Did you get an error that the server is not properly configured?
...
+
+In some deployments are SMTP servers that are not supporting authentication and TLS. In this case, you can disable authentication by not defining variable `MAIL_USERNAME` and `MAIL_PASSWORD` (which are by default `None`). If your SMTP server does not support TLS or SSL you can disable encryption by setting `MAIL_USE_TLS` and `MAIL_USE_SSL` to `false`. It's recommended to use authentication and TLS encryption.
```
Comment (alex-cit) [inline src/server/troubleshoot/index.md:29]:
```suggestion
In some deployments, there may be SMTP servers that do not support authentication and TLS. In that case, you can disable authentication by not defining variables `MAIL_USERNAME` and `MAIL_PASSWORD` (which are set to `None` by default). 

If your SMTP server does not support TLS or SSL, you can disable encryption by setting `MAIL_USE_TLS` and `MAIL_USE_SSL` to `false`. However, it is recommended to use authentication and TLS encryption.
```

File: src/server/troubleshoot/index.md [hunk 8]
```diff
@@ -17,3 +17,18 @@ Did you get an error that the server is not properly configured?
...
+
+Server is sending emails with celery worker. If you are not receiving emails, check if celery worker is running. You can check logs for sending emails in `celery-worker` container:
```
Comment (alex-cit) [inline src/server/troubleshoot/index.md:31]:
```suggestion
Server is sending emails with a celery worker. If you are not receiving emails, check if celery worker is running. Check logs for sending emails in the `celery-worker` container:
```

File: src/server/troubleshoot/index.md [hunk 9]
```diff
@@ -17,3 +17,18 @@ Did you get an error that the server is not properly configured?
    
 2. Restart the container with the `MERGIN_BASE_URL` variable
 
+## Emails are not sent
+
+If you are not receiving emails, check that the following environment variables are set correctly:
```
Comment (alex-cit) [inline src/server/troubleshoot/index.md:22]:
```suggestion
If you are not receiving emails, check that the following [environment variables](../administer/environment/) are set correctly:
```

**Review by alex-cit — COMMENTED** [review]
Great! I added some suggestions (I hope they don't change the meaning of the instructions :) )

File: src/server/administer/environment.md [hunk 10]
```diff
@@ -1,15 +1,15 @@
...
 #### Server basics
 Variables marked with star ⭐️ need to be modified for production use.
 
 | Variable name            | Type      | Default   | Description |
 |--------------------------|-----------|-----------|-------------|
-| `CONTACT_EMAIL`⭐️         | string    |           | Email contact for application administrator. |
+| `MERGIN_BASE_URL`⭐️       | string    |        | Deployment URL where <MainPlatformName /> is hosted. |
 | `COLLECT_STATISTICS`     | Boolean   | `true`    | Whether to send usage statistics for application improvements. |
-| `MERGIN_BASE_URL`⭐️       | string    |           | Deployment URL where <MainPlatformName /> is hosted. |
+| `CONTACT_EMAIL`         | string    |     ``      | Email contact for application administrator. |
```
Comment (varmar05) [inline src/server/administer/environment.md:12]: Do we still use this one, or is it some legacy variable?

File: src/server/troubleshoot/index.md [hunk 11]
```diff
@@ -1,6 +1,6 @@
 # Troubleshoot Custom Servers
 
-This article will help you debug and resolve issues in your <CommunityPlatformNameLink /> or <EnterprisePlatformNameLink /> deployment. If you use the main SaaS <DashboardLink desc="Mergin Maps Server"/>, it is always up-to-date and managed by <MainPlatformName /> team, so report your problems to us as [described here](../../misc/troubleshoot/index.md). Read more about server platforms in [overview article](../index.md).
+This article will help you debug and resolve issues in your <CommunityPlatformNameLink /> or <EnterprisePlatformNameLink /> deployment. If you use the main SaaS <DashboardLink desc="Mergin Maps Server"/>, it is always up-to-date and managed by <MainPlatformName /> team, so report your problems to us as [described here](../../misc/troubleshoot/index.md). Read more about server platforms in [overview article](../index.md). Follow [installation guide](../install/index.md) to install your own server. Documentation to environment variables and other configuration options can be found in [configure](../administer/environment.md) section.
```
[the remark this answers is not in the document]
Comment (varmar05) [reply src/server/troubleshoot/index.md:3]: I suggest to change 'SaaS' to 'Cloud'

File: src/server/troubleshoot/index.md (hunk 8 above)
Comment (varmar05) [reply src/server/troubleshoot/index.md:31]: How about "Emails are sent by celery worker container (link to docker-compose file)."

File: src/server/troubleshoot/index.md [hunk 12]
```diff
@@ -17,3 +17,18 @@ Did you get an error that the server is not properly configured?
...
+* `MAIL_DEFAULT_SENDER` is mandatory and should be a valid email address
+* `MAIL_SERVER` should be a valid url to SMTP server
+* `MAIL_PORT` if you have different port than 587, configure it with this variable
+* `MAIL_SUPPRESS_SEND` check if you set it to `False`
+
+In some deployments are SMTP servers that are not supporting authentication and TLS. In this case, you can disable authentication by not defining variable `MAIL_USERNAME` and `MAIL_PASSWORD` (which are by default `None`). If your SMTP server does not support TLS or SSL you can disable encryption by setting `MAIL_USE_TLS` and `MAIL_USE_SSL` to `false`. It's recommended to use authentication and TLS encryption.
+
+Server is sending emails with celery worker. If you are not receiving emails, check if celery worker is running. You can check logs for sending emails in `celery-worker` container:
+```shell
+$ docker logs celery-worker
```
Comment (varmar05) [inline src/server/troubleshoot/index.md:33]: Shall we add some example of what admins should expect to see?

**Review by varmar05 — CHANGES_REQUESTED** [review]

**Comment by varmar05** [comment]
@MarcelGeo let's make sure we address typical issues
- celery worker container crashing - generic config issue (e.g. RUN_CELERY legacy issues)
- celery worker process not working / stuck - generic config issue (e.g. MONKEY_PATCH issue)
- email not processed - probably not sent to queue by flask (mail config issue)
- email processed (success in logs) but no email received - SMTP config issue

File: src/server/administer/environment.md (hunk 10 above)
Comment (MarcelGeo) [reply src/server/administer/environment.md:12]: It was legacy... To improve contacting admins in deployments, we are sending It with stats

**Comment by MarcelGeo** [comment]
> @MarcelGeo let's make sure we address typical issues
> 
>     * celery worker container crashing - generic config issue (e.g. RUN_CELERY legacy issues)
> 
>     * celery worker process not working / stuck - generic config issue (e.g. MONKEY_PATCH issue)
> 
>     * email not processed - probably not sent to queue by flask (mail config issue)
> 
>     * email processed (success in logs) but no email received - SMTP config issue

added success scenario for emails. Celery troubleshooting will be another task.

**Comment by MarcelGeo** [comment]
@alex-cit review new changes and we can merge It :)

File: src/server/administer/environment.md [hunk 13]
```diff
@@ -62,14 +62,18 @@ To ease the process of permission (user) management, you can set the following g
...
-| `MERGIN_LOGO_URL`       | string    | `null`    | Link to logo in emails. |
+| `MAIL_SUPPRESS_SEND`     | Boolean   | `false`    | Whether to suppress email sending. If set to `false`, you should define the following variables. |
+| `MAIL_DEFAULT_SENDER`⭐️   | string    |           | Sender of <MainPlatformName /> emails. Best to have some no-reply address.  |
+| `MAIL_SERVER`⭐️           | string    |`localhost`| SMTP mail server host.  |
+| `MAIL_PORT`⭐️             | integer   | `587`     | SMTP mail server port.  |
+| `MAIL_USERNAME`         | string    |   `None`   | Username of user connecting to SMTP server.  |
+| `MAIL_PASSWORD`         | string    |   `None`        | Password for user connecting to SMTP server.  |
+| `MAIL_USE_TLS`🛡️        | Boolean    |   `true`        | Use TLS encryption when connecting to SMTP server.  |
+| `MAIL_USE_SSL`🛡️        | Boolean    |   `false`        | Whether to use SSL encryption when connecting to SMTP server.  |
+| `MAIL_BCC`              | string    |   `None`   | Email address to send copies of all emails.  sent. Should be system/application administrator. Mandatory in versions until 2024.4.0  |
```
Comment (alex-cit) [inline src/server/administer/environment.md:73]:
```suggestion
| `MAIL_BCC`              | string    |   `None`   | Email address to send copies of all sent emails. Should be system/application administrator. Mandatory in versions until 2024.4.0.  |
```

File: src/server/install/index.md [hunk 14]
```diff
@@ -43,4 +44,4 @@ $ docker exec merginmaps-server flask user create <username> <password> --is-adm
 
 ### Setup environment
 ​
-Now tweak deployment settings by modifying environment variables. You have to fix all variables marked as `fixme` (with asterisks) in this list of [environment variables](../administer/environment.md).
+Now tweak deployment settings by modifying environment variables. You have to fix all variables marked as required in this list of [environment variables](../administer/environment.md). In [troubleshoot](../troubleshoot/index.md) section are listed some of the most common issues with custom deployments.
```
Comment (alex-cit) [inline src/server/install/index.md:47]:
```suggestion
Now tweak deployment settings by modifying environment variables. You have to fix all variables marked as required in this list of [environment variables](../administer/environment.md). Some of the most common issues with custom deployments are listed in the [troubleshoot](../troubleshoot/index.md) section.
```

**Review by alex-cit — APPROVED** [review]
Great! I have left some suggestions

**Pull request merged**